### ISSUE TITLE
Setup staging environment

### ISSUE THREAD

**Issue by cuddle-dev** [pavlovcik/uad-ui#296]
## Problems

The dapp just interacts with mainnet contracts which will block us to test contracts interaction bc it requires real ETH. 
If we can get needed contracts deployed on testnets(goerli, optimism goerli, etc) and make the dapp working on the chains configured, we will not be blocked by the gas consumption.

## Steps to do that

- [ ] Deploy smart contracts on testnet
- [ ] Update frontend to make it working with any chains configured (maybe its already done but if not, we need to update)
- [ ] Github ci/cd should be able to deploy untrusted pull requests to staging automatically.

**Comment by 0x4007** [comment]
We already have several issues for this. Let's make the custom private testnet instead of deploying to public testnets. If for some reason we can't figure out how to make the private testnet work, then we can fall back to these options. 

#209 #193
